### PR TITLE
Adding a solution for searching within a linked list

### PR DESCRIPTION
**Pull request by kmwarter**
Checklist:

- [x ] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x ] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x ] My pull request targets the `master` branch of freeCodeCamp.
- [ x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #XXXXX

### CODE REVIEW

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@kmwarter Although your solution does pass the tests as they are currently written.  I realized after testing your code, that the current tests to not reflect the requirements of the actual challenge.   The instructions indicate adding a working `isEmpty` method, so you will need need to add a working `isEmpty` method to the solution before we can merge this PR.  Also, you need to remove the ``// solution required` comment from the solution also.

In the mean time, I will create a separate commit on this PR to add/update the tests accordingly, so we would have a valid solution and applicable working tests.

**Comment by RandellDawson** [comment]
@kmwarter I have made the necessary changes to the tests.  You just need to need to make the changes I mentioned in my first comment above.

**Comment by kmwarter** [comment]
Good catch. I added the isEmpty method too. Thanks!

**Review by RandellDawson — APPROVED** [review]
@moT01 @ojeytonwilliams Since I added a commit with test changes and new tests to this PR, can you both review this one before it gets merged?  Thanks.

**Review by moT01 — APPROVED** [review]
LGTM 🎉

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/search-within-a-linked-list.english.md [hunk 1]
```diff
@@ -22,16 +22,24 @@ Write an <code>isEmpty</code> method that checks if the linked list is empty, an
...
     testString: assert((function(){var test = new LinkedList(); return (typeof test.elementAt === 'function')}()));
-  - text: Your <code>size</code> method should return the length of the linked list
-    testString: assert((function(){var test = new LinkedList(); test.add('cat'); test.add('dog'); test.add('kitten'); return test.size() === 3}()));
-  - text: Your <code>indexOf</code> method should return the index of the given element.
-    testString: assert((function(){var test = new LinkedList(); test.add('cat'); test.add('dog'); test.add('kitten'); return test.indexOf('kitten') === 2}()));
-  - text: Your <code>elementAt</code> method should return at element at a given index.
+  - text: Your <code>elementAt</code> method should return the element found at a given index in linked list.
     testString: assert((function(){var test = new LinkedList(); test.add('cat'); test.add('dog'); test.add('kitten'); return test.elementAt(1) === 'dog'}()));
+  - text: Your <code>elementAt</code> method should return <code>undefined</code> if the given index is not found at a given index in linked list.
```
Comment (Manish-Giri) [inline curriculum/challenges/english/08-coding-interview-prep/data-structures/search-within-a-linked-list.english.md]:
```suggestion
  - text: Your <code>elementAt</code> method should return <code>undefined</code> if the given element is not found at a given index in linked list.
```

**Review by Manish-Giri — CHANGES_REQUESTED** [review]

**Comment by RandellDawson** [comment]
@Manish-Giri Thanks for catching that typo.  Looks like this one is good to be approved by you now?

**Review by Manish-Giri — APPROVED** [review]

**Comment by Manish-Giri** [comment]
@RandellDawson Yes, looks good.

**Review by moT01 — APPROVED** [review]
Congratulations on your first contribution to freeCodeCamp @kmwarter! 🎉 We look forward to future contributions. 

For future reference, it's nice if you can make your branch name a little more descriptive. Thanks and happy coding!

**Pull request merged**